README.md
自己写的工具

sqlite3数据库

域名收集：利用各种sub，oneforall，amass，接口将域名放入域名表中，目前只针对web业务
被动url爬去，再利用一个参数爬取，目录js中的目录进行提取，将结果放入xray中
url测试：当我们遇到一个有趣的url链接时，会爬取，参数发现，js目录发现


想实现src模式（上面为src模式），apt模式，收集模式


history.txt

# 2020-12-29 22:04:03.338592
+234

# 2020-12-29 22:04:05.954223
+sdg

# 2020-12-29 22:04:06.648909
+adg

# 2020-12-29 22:05:24.898154
+13

# 2020-12-29 22:05:27.401162
+admin

# 2020-12-29 22:05:30.249987
+32326

# 2020-12-29 22:05:31.385190
+8678

# 2020-12-29 22:05:36.408467
+admin

# 2020-12-29 22:05:53.072784
+exit

# 2020-12-29 22:21:07.059587
+select

# 2020-12-29 22:21:15.474144
+se

# 2020-12-29 22:21:17.346850
+select

# 2020-12-29 22:37:28.219112
+exit


test.py
import time
import subprocess

def cmd(command):
    subp = subprocess.Popen(command,shell=True,stdout=subprocess.PIPE,stderr=subprocess.PIPE,encoding="utf-8")
    subp.wait(2)
    if subp.poll() == 0:
        print(subp.communicate()[1])
    else:
        print("失败")

cmd("pwd")
cmd("exit 1")


db.py
import sqlite3
import ujson
import os
from urllib import parse
#一个url表 一个域名表 端口表（ip）

# con.execute(query)

class Ato():
    def __init__(self,domain):
        self.domains=domain
        self.sql=self.__creat_sql()
        self.jsons =os.getcwd()+"/tmp/"
        self.__oneforall(j=self.jsons)
        self.__subfinder(j=self.jsons)
    def __creat_sql(self):
        name=self.domains+".sqlite3"
        con = sqlite3.connect(name)
        query = """CREATE TABLE domain
        (id integer PRIMARY KEY AUTOINCREMENT,
        domains  varchar(30) UNIQUE,
        title varchar(20),
        code  int(5)
        );"""
        try:
            con.execute(query)
        except Exception:
            pass
        con.close()
        return name
    def __oneforall(self,j):
        con = sqlite3.connect(self.sql)
        with open("tuhu.org.json") as f:
        #with open(j+self.domains+".json") as f:
            data = ujson.load(f)
            for line in data:
                title = line['title']
                title = title.replace('\'', '')
                title = title.replace('\\n', '')
                sql = "insert into domain(domains,title,code) values('%s','%s',%d)" % (line['url'], title, line['status'])
                #         print(sql)
                #         # sql="insert into domain(domains,title,code) values('%s',%s,%d)" %(line['subdomain'],line['title'],line['status'])
                #         # print(sql)
                try:
                    con.execute(sql)
                except Exception:
                    pass
        con.commit()
        con.close()
    def __subfinder(self,j):
        con = sqlite3.connect(self.sql)
        with open("sub_tuhu.org.json", encoding='UTF-8') as  f:
        #with open(j+"sub_"+self.domains+".json",encoding='UTF-8') as  f:
            for t in f:
                data=ujson.loads(t)
                if parse.urlsplit(data['url']).port == 80:
                    data['url']=data['url'][:-3]
                if parse.urlsplit(data['url']).port == 443:
                    data['url']=data['url'][:-4]
                sql = "insert into domain(domains,title,code) values('%s','%s',%d)" % (data['url'], data['title'], data['status-code'])
                try:
                    con.execute(sql)
                except Exception:
                    pass #重复数据
        con.commit()
        con.close()
    def show(self):
        con=sqlite3.connect(self.sql)
        sql="select * from domain"
        out=con.execute(sql).fetchall()
        for t in out:
            print(t[1],"\t",t[2],t[3])
        sql="select count(*) from domain"
        out=con.execute(sql).fetchall()
        print("统计:",out[0][0])
# a=Ato(domain="tuhu.org")
# a.show()
# con=sqlite3.connect("tuhu.org.sqlite3")
# a=con.execute("select * from domain;")
# b=a.fetchall()
# for t in b:
#     print(t)
# with open("sub_tuhu.org.json",encoding='UTF-8') as  f:
#     for t in f:
#         data=ujson.loads(t)
#         print(data['url'])


tools.py
from prompt_toolkit import prompt
from prompt_toolkit.completion import WordCompleter
import os
import subprocess
import db

#import huepy

#git reset --hard
#git pull
cmdcompleter=WordCompleter(['help','domain','proxy','xray','all','exit','show'])
help_mg="""
domain  域名收集类
proxy   开启代理
xray    调用xray
exit    退出
all     随便什么url
show    当前数据
"""

the=os.getcwd()+"/tmp/"
def main():
    tu()
    while True:
        choose=prompt('mytool>',
            completer=cmdcompleter,

        )
        choose2=choose
        if choose2 == "":
            continue
        cmd=choose2.split()[0]
        if cmd == "domain":
            if len(choose2.split()) > 1:
                con1=oneforall(choose2.split()[1])
            # subfinder(choose2.split()[1])
            # cmd2=choose2.split()[1]
            # db.Ato(domain=cmd2)
        elif cmd == "help":
            print(help_mg)
        elif cmd == "exit":
            exit()
        elif cmd == "show":
            con1.show()
        else:
            print("无效命令")


def tu():
    hello="""
       **                      **
  /**                     /**
 ******  ******   ******  /**
///**/  **////** **////** /**
  /**  /**   /**/**   /** /**
  /**  /**   /**/**   /** /**
  //** //****** //******  ***
   //   //////   //////  /// 
    
    """
    print(hello)

def oneforall(ym):
    oneforal=["python3", "/root/OneForAll/oneforall.py", "--target",ym,"--fmt","json","--path",the,"run"]
    try:
        all=subprocess.Popen(oneforal,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
    except BaseException:
        print("oneforall模块出错")
        all.kill()
    # if all.wait()==0:
    #     print("运行完成!")
        #     print(all.communicate())
    js=the+"sub_"+ym+".json"
    subcmd=["subfinder","-d",ym,"-silent","-all"]
    ssub=subprocess.Popen(subcmd,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
    httpcmd=["httpx","-json","-o",js,"-ports","80,443,8000,8080,8443"]
    httpx=subprocess.Popen(httpcmd,stdin=ssub.stdout,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
    while True:
        if all.poll() == 0:
            print("完成!")
            shu=db.Ato(domain=ym)
            break
    return shu








if __name__ == '__main__':
    main()
